runtime/interpreter/decode.go

```diff
@@ -1808,38 +1808,6 @@ func (d TypeDecoder) decodeDictionaryStaticType() (StaticType, error) {
 }
 
 func (d TypeDecoder) decodeIntersectionStaticType() (StaticType, error) {
-	const expectedLength = encodedIntersectionStaticTypeLength
```

Comment:
The decoding function needs to stay backward-compatible, so old stored data can be migrated

Reply:
Is this what is proposed in the entitlements migration (https://github.com/onflow/flips/pull/95)?

It might make sense to decouple the decoding from the migration, and not perform this conversion/migration during decoding. This could be done by keeping a "restricted type" field in the interface set type, which is not used anywhere else but when decoding a restricted type inside of the state migration. We can then remove it later in a follow-up PR once the state was migrated.

Reply:
Implemented in 3fcf2e9 and 685e59d

@fxamacker could you double check that these are backwards compatible 🙏

Reply:
685e59d looks good!

3fcf2e9, backward-compatibility in CCF, is actually not needed (same for e.g. JSON) – it's an external format, we only need backward-compatibility for the internal encoding (execution state).